### PR TITLE
feat: Added access analyzer archive rules

### LINKED ISSUE #495: IAM Access Analyzers
### Describe the resource.

https://docs.aws.amazon.com/sdk-for-go/api/service/accessanalyzer/

### Use Case

Deploying IAM Access Analyzer and custom Archive Rules in AWS accounts helps account users identify external resource privileges. The Archive Rules reduce the clutter in Analyzer findings.

Knowing the configuration of Access Analyzer is key to managing AWS accounts.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amanenk**
closes #495

### CODE REVIEW

File: resources/services/accessanalyzer/analyzers.go [hunk 1]
```diff
@@ -42,103 +41,102 @@ func AccessAnalyzerAnalyzer() *schema.Table {
...
 			{
 				Name:        "name",
-				Description: "The name of the analyzer.",
+				Description: "The name of the analyzer",
 				Type:        schema.TypeString,
 			},
 			{
 				Name:        "status",
-				Description: "The status of the analyzer. An Active analyzer successfully monitors supported resources and generates new findings. The analyzer is Disabled when a user action, such as removing trusted access for AWS IAM Access Analyzer from AWS Organizations, causes the analyzer to stop generating new findings. The status is Creating when the analyzer creation is in progress and Failed when the analyzer creation has failed.",
+				Description: "The status of the analyzer. The status of the analyzer. An Active analyzer successfully monitors supported resources and generates new findings. The analyzer is Disabled when a user action, such as removing trusted access for AWS IAM Access Analyzer from AWS Organizations, causes the analyzer to stop generating new findings. The status is Creating when the analyzer creation is in progress and Failed when the analyzer creation has failed.",
```
Comment (bbernays) [inline resources/services/accessanalyzer/analyzers.go:54]: `The status of the analyzer. The status of the analyzer` duplicated

File: resources/provider/migrations/timescale/21_v0.10.12.up.sql [hunk 2]
```diff
@@ -3,5 +3,23 @@
 ALTER TABLE IF EXISTS aws_ec2_eips
     DROP CONSTRAINT aws_ec2_eips_pk;
 ALTER TABLE IF EXISTS aws_ec2_eips
-    ADD CONSTRAINT aws_ec2_eips_pk PRIMARY KEY (cq_fetch_date,account_id,allocation_id);
+    ADD CONSTRAINT aws_ec2_eips_pk PRIMARY KEY (cq_fetch_date, account_id, allocation_id);
```
Comment (bbernays) [inline resources/provider/migrations/timescale/21_v0.10.12.up.sql:6]: I think the migrations should now be `22_v0.10.13`

File: resources/services/accessanalyzer/analyzers.go [hunk 3]
```diff
@@ -266,3 +299,36 @@ func fetchAccessAnalyzerAnalyzerFindingSources(ctx context.Context, meta schema.
...
+		response, err := svc.ListArchiveRules(ctx, &config, func(options *accessanalyzer.Options) {
+			options.APIOptions = append(options.APIOptions, func(stack *middleware.Stack) error {
+				if err := stack.Initialize.Add(&awsmiddleware.RegisterServiceMetadata{
+					Region:        c.Region,
+					ServiceID:     accessanalyzer.ServiceID,
+					SigningName:   "access-analyzer",
+					OperationName: "ListFindings",
+				}, middleware.Before); err != nil {
+					return nil
+				}
```
[the remark this answers is not in the document]
Comment (roneli) [reply resources/services/accessanalyzer/analyzers.go:319]: There was an issue with access analyzer in AWS SDK it was missing a middleware, so I manually added it. Might have been fixed, we need to update the service and check

[the remark this answers is not in the document]
Comment (bbernays) [reply resources/services/accessanalyzer/analyzers.go:319]: We should document that issue so we can track it and remove this code when it is no longer needed

**Review by bbernays — APPROVED** [review]

File: resources/services/accessanalyzer/analyzers.go [hunk 4]
```diff
@@ -266,3 +289,26 @@ func fetchAccessAnalyzerAnalyzerFindingSources(ctx context.Context, meta schema.
...
+	svc := c.Services().Analyzer
+	config := accessanalyzer.ListArchiveRulesInput{
+		AnalyzerName: analyzer.Name,
+	}
+	for {
+		response, err := svc.ListArchiveRules(ctx, &config, func(options *accessanalyzer.Options) {
+			options.Region = c.Region
+		})
+		if err != nil {
+			return err
```
Comment (yevgenypats) [inline resources/services/accessanalyzer/analyzers.go:304]: Missing diag.WrapError

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Looks good. One comment + rebase needed.

File: resources/services/accessanalyzer/analyzers.go [hunk 5]
```diff
@@ -237,17 +270,7 @@ func fetchAccessAnalyzerAnalyzerFindings(ctx context.Context, meta schema.Client
...
-					OperationName: "ListFindings",
-				}, middleware.Before); err != nil {
-					return nil
-				}
-				return nil
-			})
+			options.Region = c.Region
 		})
 		if err != nil {
 			return err
```
Comment (yevgenypats) [inline resources/services/accessanalyzer/analyzers.go:276]: Add diag.WrapError

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**